### PR TITLE
Refactor property mapping creation into a class

### PR DESCRIPTION
**Pull request by aleho**
This refactors parts of the property mapping factory into a resolver to allow for decoration via an interface.

Refs #1317

### CODE REVIEW

**Comment by aleho** [comment]
I wasn't sure whether you'd also want the MappingFactory to be used as an interface, but for my use-case the resolver does everything I need.

File: src/Mapping/PropertyMappingResolver.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Vich\UploaderBundle\Exception\MappingNotFoundException;
+use Vich\UploaderBundle\Naming\ConfigurableInterface;
+use Vich\UploaderBundle\Util\ClassUtils;
+
+/**
+ * PropertyMappingResolver.
+ *
+ * @author Dustin Dobervich <ddobervich@gmail.com>
```
Comment (garak) [inline src/Mapping/PropertyMappingResolver.php:15]: Please add your references here. If you prefer not be mentioned, remove this line

**Review by garak — CHANGES_REQUESTED** [review]

File: src/Mapping/PropertyMappingResolver.php (hunk 1 above)
Comment (aleho) [reply src/Mapping/PropertyMappingResolver.php:15]: The changes are minimal, so I felt ashamed adding my name up there, but if you insist ;).

File: src/Mapping/PropertyMappingResolver.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * @internal
```
[the remark this answers is not in the document]
Comment (garak) [reply src/Mapping/PropertyMappingResolver.php:18]: The "internal" is warning developers if they use the class directly (for example, phpStorm is showing the internal class with a striking format and it's showing a warning).

File: src/Mapping/PropertyMappingResolver.php (hunk 1 above)
Comment (aleho) [reply src/Mapping/PropertyMappingResolver.php:15]: Reverted back to no reference.

**Review by garak — APPROVED** [review]

**Comment by garak** [comment]
Thanks @aleho

**Pull request merged**